### PR TITLE
{math}[foss/2017b,intel/2017b] MATLAB-Engine v2018b

### PR DESCRIPTION
**Pull request by vanzod**
Depends on:
~#7759~ (MATLAB-2018b)

### CODE REVIEW

File: easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2018b-foss-2017b-Python-2.7.14.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# step, both build and install must be performed in a single command.
+
+prebuildopts = 'cd $EBROOTMATLAB/extern/engines/python && '
+buildopts = '--build-base=%(builddir)s install --prefix=%(installdir)s'
+
+skipsteps = ['install']
+
+options = {'modulename': 'matlab.engine'}
+
+sanity_check_commands = ["python -c 'import matlab.engine; eng = matlab.engine.start_matlab(); eng.quit()'"]
```
Comment (boegel) [inline easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2018b-foss-2017b-Python-2.7.14.eb]: @vanzod This requires access to the MATLAB license server, not sure if it's a good idea to include this in the sanity check?

Fails on our workernodes with:

```
 matlab.engine.EngineError: Unable to launch MVM server: License Error
```

Comment (vanzod) [reply easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2018b-foss-2017b-Python-2.7.14.eb]: How about commenting it and leave to the user?

Comment (boegel) [reply easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2018b-foss-2017b-Python-2.7.14.eb]: Makes sense, I don't think it's a good idea to leave it in by default (and the `import` test is good enough?)

Comment (vanzod) [reply easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2018b-foss-2017b-Python-2.7.14.eb]: The import test is enough. I just wanted an additional test to ensure that the connection with MATLAB works.
I have commented it out.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2480.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a29c44f8e97a3306e85944dd5ed654e9 for a full test report.

**Comment by vanzod** [comment]
@boegel Since we established that there is no easy way to install this with `pip`, is it good to go?

**Comment by boegel** [comment]
@vanzod We'll need to fix the failing test, it's tripping over the `2018b` version for `MATLAB-Engine`...

And we'll need to add a whitelist for not using `pip` for `MATLAB-Engine`.

**Comment by boegel** [comment]
@vanzod Issue with `2018b` version triggering a false positive in dependency variants check should be fixed with https://github.com/vanzod/easybuild-easyconfigs/pull/4

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2109.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/27872e6448dcd81caebca614676e2ebb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2525.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c3e233f9d6b396537a399156be81c15a for a full test report.

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**